Formula/v/vcsh.rb

```diff
@@ -1,14 +1,19 @@
 class Vcsh < Formula
   desc "Config manager based on git"
   homepage "https://github.com/RichiH/vcsh"
-  url "https://github.com/RichiH/vcsh/releases/download/v2.0.8/vcsh-2.0.8.tar.xz"
-  sha256 "560440defe4f20ac22ce65e873c7ff60ca0c08318524afe6dae86adc4b13d714"
+  url "https://github.com/RichiH/vcsh/releases/download/v2.0.10/vcsh-2.0.10.tar.zst"
+  sha256 "6ed8f4eee683f2cc8f885b31196fdc3b333f86ebc3110ecd1bcd60dfac64c0b4"
   license "GPL-2.0-or-later"
 
   bottle do
     sha256 cellar: :any_skip_relocation, all: "d7917b8660f585cc6bf699f74d25d920d855390e2ea526fa27fb76ab26159e53"
   end
 
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "bash-completion"
```

Comment:
It's weird to have `bash-completion` as a runtime dependency. Is it needed for the software to function?

If it's only needed for completions, I think we should leave it to the user to install it.

Reply:
trying to fix this issue

```
  checking for a sed that does not truncate output... (cached) sed
  ./configure: line 3148: syntax error near unexpected token `BASH_COMPLETION,'
  ./configure: line 3148: `  PKG_CHECK_MODULES(BASH_COMPLETION, bash-completion >= 2.0,'
```

Reply:
I can build without `bash-completion`:

<details>

```console
linuxbrew@c3658bf814fb:~/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core$ git diff
diff --git a/Formula/v/vcsh.rb b/Formula/v/vcsh.rb
index b8b0378800b..719e2b1f3ff 100644
--- a/Formula/v/vcsh.rb
+++ b/Formula/v/vcsh.rb
@@ -1,14 +1,18 @@
 class Vcsh < Formula
   desc "Config manager based on git"
   homepage "https://github.com/RichiH/vcsh"
-  url "https://github.com/RichiH/vcsh/releases/download/v2.0.8/vcsh-2.0.8.tar.xz"
-  sha256 "560440defe4f20ac22ce65e873c7ff60ca0c08318524afe6dae86adc4b13d714"
+  url "https://github.com/RichiH/vcsh/releases/download/v2.0.10/vcsh-2.0.10.tar.zst"
+  sha256 "6ed8f4eee683f2cc8f885b31196fdc3b333f86ebc3110ecd1bcd60dfac64c0b4"
   license "GPL-2.0-or-later"

   bottle do
     sha256 cellar: :any_skip_relocation, all: "d7917b8660f585cc6bf699f74d25d920d855390e2ea526fa27fb76ab26159e53"
   end

+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+
   def install
     # Set GIT, SED, and GREP to prevent
     # hardcoding shim references and absolute paths.
linuxbrew@c3658bf814fb:~/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core$ brew install -sv vcsh
==> Auto-updating Homebrew...
Adjust how often this is run with HOMEBREW_AUTO_UPDATE_SECS or disable with
HOMEBREW_NO_AUTO_UPDATE. Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
==> Fetching vcsh
==> Downloading https://github.com/RichiH/vcsh/releases/download/v2.0.10/vcsh-2.0.10.tar.zst
Already downloaded: /home/linuxbrew/.cache/Homebrew/downloads/ed73ef43d4f1c19d14d304cbf872fd281a84e404a9738946fb53eea024c7be30--vcsh-2.0.10.tar.zst
/usr/bin/env PATH=/home/linuxbrew/.linuxbrew/opt/zstd/bin:/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/shims/linux/super:/home/linuxbrew/.linuxbrew/opt/autoconf/bin:/home/linuxbrew/.linuxbrew/opt/automake/bin:/home/linuxbrew/.linuxbrew/opt/libtool/bin:/usr/bin:/bin:/usr/sbin:/sbin unzstd -T0 --rm -- /tmp/homebrew-tar20240413-8625-4ei3tr/ed73ef43d4f1c19d14d304cbf872fd281a84e404a9738946fb53eea024c7be30--vcsh-2.0.10.tar.zst
/tmp/homebrew-tar20240413-8625-4ei3tr/ed73ef43d4f1c19d14d304cbf872fd281a84e404a9738946fb53eea024c7be30--vcsh-2.0.10.tar.zst: 491520 bytes
/usr/bin/env tar --extract --no-same-owner --file /tmp/homebrew-tar20240413-8625-4ei3tr/ed73ef43d4f1c19d14d304cbf872fd281a84e404a9738946fb53eea024c7be30--vcsh-2.0.10.tar --directory /tmp/homebrew-unpack20240413-8625-m4ng3b
/usr/bin/env cp -pR /tmp/homebrew-unpack20240413-8625-m4ng3b/vcsh-2.0.10/. /tmp/vcsh-20240413-8625-krbiih/vcsh-2.0.10
==> ./configure --with-zsh-completion-dir=/home/linuxbrew/.linuxbrew/Cellar/vcsh/2.0.10/share/zsh/site-functions --with-bash-completion-dir=/home/linuxbrew/.linuxbrew/Cellar/vcsh/2.0.10/etc/bash_completion.d GIT=git SED=sed GREP=grep --disable-debug --disable-dependency-tracking --prefix=/home/linuxbrew/.linuxbrew/Cellar/vcsh/2.0.10 --libdir=/home/linuxbrew/.linuxbrew/Cellar/vcsh/2.0.10/lib
configure: WARNING: unrecognized options: --disable-debug, --disable-dependency-tracking
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking how to create a pax tar archive... gnutar
checking whether make supports nested variables... (cached) yes
checking for a sed that does not truncate output... sed
checking whether to enable developer-specific portions of Makefiles... no
configure: adding automake macro support
configure: creating aminclude.am
checking for gawk... (cached) gawk
checking for grep that handles long lines and -e... grep
cat: build-aux/que_git_version.am: No such file or directory
checking for pkg-config... no
checking for a sed that does not truncate output... (cached) sed
checking for bash-completion >= 2.0... no
cat: build-aux/que_shell_completion_dirs.am: No such file or directory
checking for grep that handles long lines and -e... (cached) grep
checking for a sed that does not truncate output... (cached) sed
checking for comm... /usr/bin/comm
checking for wc... /usr/bin/wc
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating vcsh
config.status: creating doc/vcsh.1.ronn
configure: WARNING: unrecognized options: --disable-debug, --disable-dependency-tracking
==> make install
 cd . && /bin/bash /tmp/vcsh-20240413-8625-krbiih/vcsh-2.0.10/build-aux/missing automake-1.16 --foreign Makefile
aclocal.m4:17: warning: this file was generated for autoconf 2.71.
You have another version of autoconf.  It may work, but is not guaranteed to.
If you have problems, you may need to regenerate the build system entirely.
To do so, use the procedure documented by the package, typically 'autoreconf'.
 cd . && /bin/bash ./config.status Makefile
config.status: creating Makefile
/usr/bin/make  install-am
make[1]: Entering directory '/tmp/vcsh-20240413-8625-krbiih/vcsh-2.0.10'
make[2]: Entering directory '/tmp/vcsh-20240413-8625-krbiih/vcsh-2.0.10'
 /usr/bin/mkdir -p '/home/linuxbrew/.linuxbrew/Cellar/vcsh/2.0.10/bin'
 /usr/bin/mkdir -p '/home/linuxbrew/.linuxbrew/Cellar/vcsh/2.0.10/share/doc/vcsh'
 /usr/bin/mkdir -p '/home/linuxbrew/.linuxbrew/Cellar/vcsh/2.0.10/share/licenses/vcsh'
 /usr/bin/mkdir -p '/home/linuxbrew/.linuxbrew/Cellar/vcsh/2.0.10/share/doc/vcsh/sample_hooks'
 /usr/bin/mkdir -p '/home/linuxbrew/.linuxbrew/Cellar/vcsh/2.0.10/share/man/man1'
 /usr/bin/install -c -m 644 changelog doc/INSTALL.md doc/README.md doc/error_codes.md '/home/linuxbrew/.linuxbrew/Cellar/vcsh/2.0.10/share/doc/vcsh'
 /usr/bin/install -c vcsh '/home/linuxbrew/.linuxbrew/Cellar/vcsh/2.0.10/bin'
 /usr/bin/install -c -m 644 doc/sample_hooks/post-init-add-origin doc/sample_hooks/post-init-setup-mr doc/sample_hooks/post-merge-unclobber doc/sample_hooks/pre-merge-unclobber '/home/linuxbrew/.linuxbrew/Cellar/vcsh/2.0.10/share/doc/vcsh/sample_hooks'
 /usr/bin/install -c -m 644 LICENSE.md CONTRIBUTORS '/home/linuxbrew/.linuxbrew/Cellar/vcsh/2.0.10/share/licenses/vcsh'
 /usr/bin/install -c -m 644 doc/vcsh.1 '/home/linuxbrew/.linuxbrew/Cellar/vcsh/2.0.10/share/man/man1'
make[2]: Leaving directory '/tmp/vcsh-20240413-8625-krbiih/vcsh-2.0.10'
make[1]: Leaving directory '/tmp/vcsh-20240413-8625-krbiih/vcsh-2.0.10'
==> Cleaning
==> Finishing up
==> Summary
🍺  /home/linuxbrew/.linuxbrew/Cellar/vcsh/2.0.10: 17 files, 164.4KB, built in 2 seconds
==> Running `brew cleanup vcsh`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
```
</details>

Reply:
The issue you encountered was a5baf1e95de04473a73403bc5494d37bd42d8e74, but you've removed the `autoreconf` step too. I think the problem is with `autoreconf` somehow, though I cannot determine exactly.

Also, `bash-completion` is on 1.3 so it doesn't make much sense if adding `bash-completion` fixes this.

Reply:
make sense, let me remove it.